### DISCUSSION TITLE
Special character and Emoji Picker

### DISCUSSION THREAD

**Discussion by 5-pebbles** [helix-editor/helix#11289] [Ideas] [answered]
I would like something like this in Helix: https://github.com/nvim-telescope/telescope-symbols.nvim
> Though, this may be a better fit for a plugin.

At least searching and inserting emojis at the cursor would be very nice.
I would be happy to figure out the implementation and make a PR, but I am not sure what would be best to include. So I am looking for ideas and opinions.

**Comment by the-mikedavis** [comment, marked as answer]
IMO this does not belong in core, it should be done as a plugin instead when it becomes possible.

Also though I think this is better solved at the OS level. For example I use https://github.com/Zeioth/wofi-emoji. I believe macOS also has an emoji tool that works in a similar way.

**Reply by kirawi** [reply]
This could also be done at the terminal level, IIRC Kitty has something like this

**Reply by 5-pebbles** [reply]
I would still like it in my editor, but you are right most people would already be using one of the other systems. Therefore, it would be better as a plugin. Thank you very much for your time!

**Reply by kirawi** [reply]
For something more general, you might be interested in https://github.com/helix-editor/helix/issues/1438. The idea so far was that it would function as a special extension of snippets.

**Comment by MrDowntempo** [comment]
I found this thread when looking for an emoji picker plugin. I agree it probably should be a plugin, and not in core. But I DO really want one. Using a separate OS level tool, means I have to leave my editor to get the emoji and then come back. The snippet solution is interesting, but not as comprehensive as an emoji selector. Might work if you reuse a couple over and over, but not if you want to pull one you've never used before or use rarely.

**Comment by felix-as** [comment]
Symbols, Emojis and "special character" are often needed or at least helpful when editing text (and I guess it will only get more). I'd prefer to be able to use a builtin-solution so that my editor is a "go-to" solution if i want.
The OS might provide something and the editor will surely not prevent me using that one if i want. I have not seen these providing a better solution than the telescope-symbols referenced bu the OT. As this is "just" another picker, it weakens "should not be in core" arguments. Probably low cost, but high gain and a killer marketing gimmick for screenshots :D .
